oteapi/models/secretconfig.py

```diff
@@ -0,0 +1,67 @@
+"""AttrDict for specifying user credentials or secrets."""
+
+import json
+from typing import Optional
+
+from pydantic import Field, SecretStr
+
+from oteapi.models.genericconfig import AttrDict
+from oteapi.settings import settings
+
+
+def json_dumps(model, *, default):
+    """Alternative function for dumping exposed
+    secrets to json when model is serialized."""
+    return json.dumps(
+        {
+            key: (
+                value.get_secret_value()
+                if settings.expose_secrets and isinstance(value, SecretStr)
+                else value
+            )
+            for key, value in model.items()
+        },
+        default=default,
+    )
+
+
+class SecretConfig(AttrDict):
+    """Simple model for handling secret in other config-models."""
+
+    user: Optional[SecretStr] = Field(None, description="User name for authentication.")
+    password: Optional[SecretStr] = Field(
+        None, description="Password for authentication."
+    )
+    token: Optional[SecretStr] = Field(
+        None,
+        description="An access token for providing access and meta data to an application.",
+    )
+    client_id: Optional[SecretStr] = Field(
+        None, description="Client ID for an OAUTH2 client."
+    )
+    client_secret: Optional[SecretStr] = Field(
+        None, description="Client secret for an OAUTH2 client."
+    )
+
+    class Config:
+        """Pydantic donfiguration for SecretConfig when serialized or exported."""
+
+        # json_encoders = {
+        #     SecretStr: lambda secret: (
+        #         secret.get_secret_value()
+        #         if settings.expose_secrets and secret
+        #         else secret
+        #     )
+        # }
+        #
+        # Using the json_encoders directly throws an
```

Comment:
I'll take a look at this as discussed in a meeting. Feel free to ignore the parts/all of this review until then.

Reply:
Let's just use `json_dumps`. I think it's elegant enough :)

Reply:
Nice! I was not aware that you can provide kwargs in the class-definition 🤔